src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -102,7 +121,20 @@ protected LogCollector logCollector() {
     public void before() throws IOException, InterruptedException {
         log.debug("Starting docker-compose cluster");
         dockerCompose().build();
-        dockerCompose().up();
+
+        try {
+            dockerCompose().up();
+        } catch (DockerExecutionException e) {
+            Set<String> conflictingContainerNames = getConflictingContainerNames(e.getMessage());
+            if (removeConflictingContainersOnStartup() && !conflictingContainerNames.isEmpty()) {
+                // if rule is configured to remove containers with existing name on startup and conflicting containers
```

Comment:
Rather than a comment this seems like an excellent place for a log line, if somebody might be confused about where their containers have gone on a CI server logging is much more discoverable.